Modules/Scripted/Home/HomeLib/dependency_installer.py

```diff
@@ -56,7 +62,14 @@ def monai_pre_install():
     slicer.util._executePythonModule('light_the_torch', ['install', 'monai'])
 
 
+cudaVersion = os.popen("nvcc --version | egrep -o 'release [0-9\.]+' | egrep -o '[0-9\.]+' | sed -r -e 's/\.//g'").read().splitlines()[0]
+print(f"{cudaVersion = }")
+
 check_and_install_itk = functools.partial(check_and_install_package, ["itk"], "itk")
 check_and_install_pandas = functools.partial(check_and_install_package, ["pandas"], "pandas")
 check_and_install_matplotlib = functools.partial(check_and_install_package, ["matplotlib"], "matplotlib")
-check_and_install_monai = functools.partial(check_and_install_package, ["monai", "skimage", "tqdm"], "monai[skimage,tqdm]", monai_pre_install)
+check_and_install_monai = functools.partial(
+    check_and_install_package,
+    ["monai", "skimage", "tqdm", "PIL", "ignite", "gdown", "nibabel", "monai.transforms", "monai.deploy.core", "scipy", "numpy"],
+    f"monai[skimage,tqdm,pillow,ignite,gdown,nibabel,transformers] monai-deploy-app-sdk scipy numpy --extra-index-url https://download.pytorch.org/whl/cu{cudaVersion}",
```

Comment:
We shouldn't altogether remove the option to use light-the-torch. The advantage of light-the-torch is that it insulates us from having to worry about how to find exactly the right pip install command for the user's environment. And if things change then maintenance becomes a lot easier because we just have to keep up with light-the-torch changes and not worry about changes form nvidia/pytorch/pip.

Maybe we can have two side-by-side buttons for installing monai: one for the light-the-torch approach and one for the direct approach. The direct approach could even let you just type in a cuda version, so that nvcc is not needed. Just throwing ideas out-- what do you think?

These installer buttons are temporary anyway... they really ought to be replaced by an installation that takes place during the build process. I just don't know cmake well enough to achieve that at the moment :)

Reply:
I think the use of `nvcc` is an experiment that failed.  I've reverted to `light-the-torch`.